lib/validator/validator.go

```diff
@@ -286,16 +285,28 @@ func validateHostEndpointSpec(v *validator.Validate, structLevel *validator.Stru
 	}
 }
 
-func validatePoolMetadata(v *validator.Validate, structLevel *validator.StructLevel) {
-	pool := structLevel.CurrentStruct.Interface().(api.IPPoolMetadata)
+func validatePool(v *validator.Validate, structLevel *validator.StructLevel) {
+	pool := structLevel.CurrentStruct.Interface().(api.IPPool)
+
+	// Metadata values may be nil when doing a list operation, so skip
+	// validation if not assigned.
+	if pool.Metadata.CIDR.IP == nil {
+		return
+	}
+
+	// IPIP cannot be enabled for IPv6.
+	if pool.Metadata.CIDR.Version() == 6 && pool.Spec.IPIP != nil && pool.Spec.IPIP.Enabled {
+		structLevel.ReportError(reflect.ValueOf(pool.Spec.IPIP.Enabled),
+			"IPIP.Enabled", "", reason("IPIP is not supported on an IPv6 IP pool"))
+	}
 
-	// The Calico IPAM places restrictions on the minimum IP pool size, check that the
-	// pool is at least the minimum size.
-	if pool.CIDR.IP != nil {
-		ones, bits := pool.CIDR.Mask.Size()
-		log.Debugf("Pool CIDR: %s, num bits: %d", pool.CIDR, bits-ones)
+	// The Calico IPAM places restrictions on the minimum IP pool size.  If
+	// the pool is enabled, check that the pool is at least the minimum size.
+	if !pool.Spec.Disabled {
```

Comment:
@caseydavenport asked for that additional change at the last minute :-)

Since there is cross Spec/Meta checking it didn't make sense to keep the PoolMetadata check.
We are still checking CIDR.IP - just that I moved it to the start of the validation method returning if the IP is nil.  On hindsight I prefer encasing the checks in that branch so I'll change it back - I think it's clearer that way and it's less error prone if people add more checks in the future.